src/main/webapp/app/shared/metis/posting-reactions-bar/posting-reactions-bar.component.ts

```
@@ -0,0 +1,484 @@
+     */
+    buildReactionMetaDataMap(reactions: Reaction[]): ReactionMetaDataMap {
+        return reactions.reduce((metaDataMap: ReactionMetaDataMap, reaction: Reaction) => {
+            const hasReacted = reaction.user?.id === this.metisService.getUser().id;
+            // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+            const reactingUser = hasReacted ? PLACEHOLDER_USER_REACTED : reaction.user?.name!;
+            const reactionMetaData: ReactionMetaData = {
+                count: metaDataMap[reaction.emojiId!] ? metaDataMap[reaction.emojiId!].count + 1 : 1,
+                hasReacted: metaDataMap[reaction.emojiId!] ? metaDataMap[reaction.emojiId!].hasReacted || hasReacted : hasReacted,
+                reactingUsers: metaDataMap[reaction.emojiId!] ? metaDataMap[reaction.emojiId!].reactingUsers.concat(reactingUser) : [reactingUser],
+            };
+            return { ...metaDataMap, [reaction.emojiId!]: reactionMetaData };
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)